src/controllers/team/delete.ts

```diff
@@ -0,0 +1,30 @@
+import * as Interfaces from "@interfaces";
+import * as Errors from "@errors";
+import { prisma } from "@utils/prisma";
+import * as Utils from "@utils";
+
+const deleteTeamById: Interfaces.Controller.Async = async (req, res, next) => {
+  const { teamId: TID } = req.params;
+  const teamId = String(TID);
+
+  if (!teamId || teamId.length !== 24) return next(Errors.Module.invalidInput);
+
+  if (!(await prisma.team.findFirst({ where: { id: teamId } })))
+    return next(Errors.Team.teamNotFound);
+
+  let deletedMembers, deletedTeam;
+  await prisma.$transaction(async (prisma) => {
+    deletedMembers = await prisma.teamRegistration.deleteMany({
+      where: { teamId: teamId },
+    });
+
+    deletedTeam = await prisma.team.delete({ where: { id: teamId } });
+  });
```

Comment:
Since, the delete behavior is set to cascade for **team registrations**. I believe it is enough to delete the **team** for the **team registrations** associated with it to be deleted and there is no need to delete them separatly. Can you confirm this and change it accordingly.